_posts/2020-11-08-basic-coding-applications-for-commercial-strategy.md

```diff
@@ -0,0 +1,83 @@
+
+#### Code to Separate Data by Territory:
+We start off by creating a general function to subset any territory out of the full data set. This code should not have to change when different files and territories are indicated. This should also create an output CSV file that is named based on the defined territory.
+```ruby
+subset_data_to_territory <- function(input_file_name, territory_to_subset) {
+
+  # read in the complete csv files
+  all_revenue_data <- readr::read_csv(input_file_name)
+
+  # subset the data set to only include rows where the territory column has the
+  # chosen territory name(s) in it
+  territory_data <- all_revenue_data %>%
+    dplyr::filter(`territory` %in% territory_to_subset)
+
+  # check that the subsetted data actually has data in it
+   if (nrow(territory_data) == 0) {
+    stop("ERROR, no rows matching given territory name. Did you make a typo?")
+   }
+
+  # save the territory list as a string to use in the file name
+  dat <- data.frame(Col2 = territory_to_subset)
+  print(head(dat))
+  territory_summary <- dat %>%
+    dplyr::summarise(Col2 = paste(Col2, collapse = "_"))
+  print(head(territory_summary))
+
+  # save the territory data to a new csv file in your output directory
+  readr::write_csv(territory_data, paste0("output/",
+                                          tools::file_path_sans_ext(
+                                            basename(input_file_name)),
+                                          "_",
+                                          territory_summary,
+                                          ".csv"))
+  return(territory_data)
+}
+
+```
+It is important that the appropriate libraries be loaded in prior to running the function.
+```ruby
+library("dplyr")
+library("readr")
+```
+Next you want to define your inputs. The below example is for the North West region that includes the states of WA, OR, NV, ID, MT, UT, WY, CO, ND, SD, NE, KS, OK, MN, and WI. The raw data file in this example is titled `name_of_invoices_report.csv`.
+```ruby
+input_file_name = paste0("data/name_of_invoices_report.csv")
+territory_to_subset = c("WA", "OR", "NV", "ID", "MT", "UT", "WY", "CO",
+                               "ND", "SD", "NE", "KS", "OK", "MN", "WI")
+```
+Finally, it's time to execute the function.
+```ruby
+subset_data_to_territory(input_file_name, territory_to_subset)
+
+```
+For this example, our result is a csv file with the name `name_of_invoices_report_WA_OR_NV_ID_MT_UT_WY_CO_ND_SD_NE_KS_OK_MN_WI.csv`. This file contains all of the revenue data from the original file for just the indicated states. With this subsetted data, you can dive deeper into territory analysis by looking at the revenue over time. This general function is a helpful tool in that it can allow you to define what `territory` is. This general variable can be defined as the city, manager, region, and more. To toggle between these different categories, simply change the chosen column name in the raw data set to `territory`.
```

Comment:
Great job on the blog post and I appreciate the commenting throughout, it really helps orient the first time user. Making the script flexible to take in a diverse input, is also really useful.